### PR TITLE
generalize splattable lists

### LINKED ISSUE #9239: Improve "splat" list errors
Currently:
```
In [15]: p.grid.grid_line_alpha
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-15-e1510245d5b4> in <module>
----> 1 p.grid.grid_line_alpha

AttributeError: '_list_attr_splat' object has no attribute 'grid_line_alpha'
```
This could be improved by clearly stating that the error is because there is more than one grid, and give a snippet of how individual values can be accessed.

### PR DESCRIPTION
**Pull request by bryevdv**
- [x] issues: fixes #9239
- [x] tests added / passed
- [x] release document entry (if new feature or API change)

cc @bokeh/dev for any comments. Seems like it should be uncontroversial so I will plan to merge tomorrow without comment.

### CODE REVIEW

File: bokeh/models/plots.py [hunk 1]
```diff
@@ -737,6 +737,15 @@ class _list_attr_splat(list):
...
         for x in self:
             setattr(x, attr, value)
+    def __getattribute__(self, attr):
+        if len(self) == 0:
+            raise AttributeError("Trying to access %r attribute on an empty 'splattable' list" % attr)
+        if len(self) == 1:
+            return getattr(self[0], attr)
+        try:
+            return _list_attr_splat([getattr(x, attr) for x in self])
+        except:
```
Comment (mattpap) [inline bokeh/models/plots.py]: `except Exception` is the preferred form. Otherwise system exceptions are going to be caught here as well.

Comment (bryevdv) [reply bokeh/models/plots.py]: Oh, thank you (coding in a hurry)

Comment (mattpap) [reply bokeh/models/plots.py]: If there is a specific exception to consider (`AttributeError`?), that would be even better.

**Comment by mattpap** [comment]
LGTM.

**Comment by philippjfr** [comment]
This is causing some regressions in HoloViews since it used to be possible to use list-like methods like `index` and `pop`, which doesn't seem to be the case anymore.

**Comment by philippjfr** [comment]
Sorry I didn't review this in time, I was traveling unfortunately.

**Comment by philippjfr** [comment]
Btw, if you decide that breaking that is worth it I won't object I just wanted to point out that this does break things which used to work.

**Comment by bryevdv** [comment]
> This is causing some regressions in HoloViews since it used to be possible to use list-like methods like index and pop, which doesn't seem to be the case anymore.

I think that's fixable by checking to see if the attribution exists on `list`. I'd prefer to fix than revert but since you have the test case code it would be best if you make the PR. Otherwise we can revert.

**Comment by philippjfr** [comment]
>I think that's fixable by checking to see if the attribution exists on list. I'd prefer to fix than revert but since you have the test case code it would be best if you make the PR. Otherwise we can revert.

+1, I'll make a PR shortly.

**Pull request merged**